src/main/java/org/zalando/nakadi/repository/db/SubscriptionDbRepository.java

```diff
@@ -56,21 +59,45 @@ public Subscription createSubscription(final SubscriptionBase subscriptionBase)
             throw new InternalNakadiException("Serialization problem during persistence of event type", e);
         } catch (final DuplicateKeyException e) {
             throw new DuplicatedSubscriptionException("Subscription with the same key properties already exists", e);
+        } catch (final DataAccessException e) {
+            throw new ServiceUnavailableException("Error occurred when running database request", e);
         }
     }
 
-    public Subscription getSubscription(final String id) throws NoSuchSubscriptionException {
+    public Subscription getSubscription(final String id) throws NoSuchSubscriptionException,
+            ServiceUnavailableException {
         final String sql = "SELECT s_subscription_object FROM zn_data.subscription WHERE s_id = ?";
         try {
             return jdbcTemplate.queryForObject(sql, new Object[]{id}, rowMapper);
         } catch (final EmptyResultDataAccessException e) {
             throw new NoSuchSubscriptionException("Subscription with id \"" + id + "\" does not exist", e);
+        } catch (final DataAccessException e) {
+            throw new ServiceUnavailableException("Error occurred when running database request", e);
+        }
+    }
+
+    public List<Subscription> listSubscriptions() throws ServiceUnavailableException {
+        try {
+            return jdbcTemplate.query("SELECT s_subscription_object FROM zn_data.subscription", rowMapper);
```

Comment:
Yes, maybe we could enforce an ordering that we think is most convenient. Creation time descending?